src/main/java/com/example/demo/model/ShoppingCart.java

```diff
@@ -0,0 +1,29 @@
+package com.example.demo.model;
+
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.Table;
+import java.util.Set;
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity
+@Getter
+@Setter
+@Table(name = "shopping_carts")
+public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToOne
+    @JoinColumn(name = "user_id", nullable = false)
+    private User user;
```

Comment:
ShoppingCart record can't exist without User, so we can make primary key of "shopping carts" table simultaneously a foreign key to the "users" table. It will help us to avoid extra column + index, because we always want to have an index for foreign key columns.
In code it will look like this (add @mapsid annotation + remove @GeneratedValue):
```suggestion
    @Id
    private Long id;
    @MapsId
    @OneToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
```